Server/src/controllers/DataSetController.ts

```diff
@@ -0,0 +1,245 @@
+import { Request, Response } from 'express';
+import { DataSetService } from '../services/DataSetService';
+import { IDataRequestModel } from "../models/interfaces/DataRequestModelInterface";
+
+export class DataSetController {
+    private processedRequest: IDataRequestModel;
+    private dataSetService: DataSetService
+
+    constructor() {
+    }
+
+    /**
+     * This controller will take a request, verify that it is valid, and if it valid will then process the request,
+     * send it to the getDataService to acquire an array of data sets that match the entered search terms, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForData(request: Request, response: Response) {
+        let validateData = this.validateInputData(request)
+        if (!validateData) {
+            response.status(400).json("Invalid search params entered");
+        }
+        else {
+            this.dataSetService = new DataSetService();
+            let requestParams: any = { ...request.query };
+            this.processedRequest = requestParams;
+            try {
+                let arrayOfData = await this.dataSetService.getArrayOfDatasets(this.processedRequest)
+                return response.status(200).send(arrayOfData);
+            } catch (err) {
+                response.status(400).send(err);
+            }
+        }
+    }
+
+    async createRequestToDeleteSavedDataSet(request: Request, response: Response) {
+        let requestParam = request.params.datasetId;
+        let datasetId = Number(requestParam);
+        if (isNaN(datasetId)) {
+            response.status(400).json("Invalid data set ID entered");
+        }
+        else {
+            let userId: number = request.body.user.account_id
+            try {
+                let requestResponse = await this.dataSetService.removeSavedDatasetService(userId, datasetId)

```

Comment:
imo saved is redundant, since it is being removed from DB, obviously it is saved, also, the method name says removeSavedDatasetService, does it remove a dataset or does it remove a service?

Reply:
Yeah. Good catch on some of the naming.... Will change

Reply:
SavedDatasets should all be renamed to UserFavoriteDatasets or similar, once some pending changes are pushed